--- a/lmdb.spec
+++ b/lmdb.spec
@@ -1,5 +1,6 @@
-%define major 0
-%define libname %mklibname %{name} %{major}
+%define major 1
+%define oldlibname %mklibname %{name} 0
+%define libname %mklibname %{name}
 %define devname %mklibname %{name} -d
 
 Summary:	Memory-mapped key-value database
@@ -31,6 +32,7 @@
 %package -n %{libname}
 Summary:	Shared library for %{name}
 Group:		System/Libraries
+Obsoletes:	%{oldlibname} < %{EVRD}
 
 %description -n %{libname}
 Shared library for %{name}.
@@ -64,7 +66,7 @@
 %build
 %set_build_flags
 cd libraries/lib%{name}
-%make_build CC=%{__cc} XCFLAGS="%{optflags} -O3" LDFLAGS="%{build_ldflags}"
+%make_build CC="%{__cc}" XCFLAGS="%{optflags} -O3" LDFLAGS="%{build_ldflags}"
 
 %install
 cd libraries/lib%{name}
